crates/red_knot_python_semantic/src/types/narrow.rs

```diff
@@ -353,7 +353,9 @@ impl<'db> NarrowingConstraintsBuilder<'db> {
                     let to_constraint = match function {
                         KnownConstraintFunction::IsInstance => {
                             |class_literal: ClassLiteralType<'db>| {
-                                Type::anonymous_instance(class_literal.class)
+                                Type::Instance(InstanceType {
+                                    class: class_literal.class,
+                                })
```

Comment:
We do this a lot in this PR, I wonder about a `Class::to_instance_type()` method for it?

Reply:
This can be considered separately as a follow-up if it's worth it

Reply:
https://github.com/astral-sh/ruff/pull/14215